### PR TITLE
start trying to use dep

### PR DESCRIPTION
**Pull request by deads2k**
@sttts I'm trying to use k8s.io/kube-gen, but I'm having trouble getting it to use a different.  Can you see if see what I'm doing wrong (just `dep ensure`)?  I'd really like to start using this, but if I can't use the dependency, I will have to hack something together on godeps+magic I think.

### CODE REVIEW

File: Gopkg.lock [hunk 1]
```diff
@@ -0,0 +1,249 @@
+# This file is autogenerated, do not edit; changes may be undone by the next 'dep ensure'.
+
+
+[[projects]]
```
Comment (sttts) [inline Gopkg.lock]: Am I the only who thinks toml is ugly?

File: Gopkg.toml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+required = [
+  "github.com/openshift/origin/pkg/template/apis/template",
+  "github.com/openshift/origin/pkg/template/apis/template/v1",
+  "k8s.io/kube-gen",
+]
+
+[[constraint]]
+  name = "k8s.io/kube-gen"
+  branch = "master"
+  source = "github.com/deads2k/kube-gen"
```
Comment (sttts) [inline Gopkg.toml]: didn't you call it fake-kube-gen?

**Comment by sttts** [comment]
Opened a golang/dep issue: https://github.com/golang/dep/issues/990

File: Gopkg.lock (hunk 1 above)
Comment (tnozicka) [inline Gopkg.lock]: count me in :)

**Comment by deads2k** [comment]
@sttts thanks for getting everything pushed out.  I've been working on this a bit at a time and I think I have something worth looking at again.

@sttts @smarterclayton I don't know why `dep` wants to pull in every bit of every repo (like `glide`)

@openshift/api-review I reorganized this into `github.com/openshift/client-go/<shortGroupName>/{clientset|internalclientset|listers|informer}`.  I figured I'd put other code under `pkg` and we'll just ban anyone from making a group named `pkg`.

**Comment by smarterclayton** [comment]
I think that's reasonable (anyone creating a `pkg` group would fail API review on not using acronyms)

File: Gopkg.toml [hunk 3]
```diff
@@ -33,6 +33,13 @@ required = [
   name = "k8s.io/code-generator"
   branch = "master"
 
+[[constraint]]
+  name = "k8s.io/apimachinery"
+  branch = "release-1.7"
```
Comment (sttts) [inline Gopkg.toml]: Note that this file misses all revision constraints for dependencies (which you only have in Gopkg.lock, which is ignored completely by dep when importing your new openshift/client-go into another project). This is even worse for transitive dependencies. There is no way at all to specify them in your Gopkg.toml for 3rdparty consumers (dep will just ignore them as they are no direct imports). This means that users cannot vendor this (without tons of manual overrides) at all. https://github.com/golang/dep/issues/1124

**Comment by sttts** [comment]
Note that Gopkg.tml misses all revision constraints for dependencies (which you only have in Gopkg.lock, which is ignored completely by dep when importing your new openshift/client-go into another project). This is even worse for transitive dependencies. There is no way at all to specify them in your Gopkg.toml for 3rdparty consumers (dep will just ignore them as they are no direct imports). This means that users cannot vendor this (without tons of manual overrides) at all. golang/dep#1124

**Comment by deads2k** [comment]
> Note that Gopkg.tml misses all revision constraints for dependencies (which you only have in Gopkg.lock, which is ignored completely by dep when importing your new openshift/client-go into another project). This is even worse for transitive dependencies. There is no way at all to specify them in your Gopkg.toml for 3rdparty consumers (dep will just ignore them as they are no direct imports). This means that users cannot vendor this (without tons of manual overrides) at all. golang/dep#1124

Soo..... don't use `dep`.  Well.  That's frustrating.

File: Gopkg.lock (hunk 1 above)
Comment (sdboyer) [reply Gopkg.lock]: i hate ALL file formats. roughly equally, and for different reasons! 😜

**Pull request closed without merging**